corehq/tests/test_middleware.py

```diff
@@ -241,3 +248,48 @@ def test_already_secure_cookie_remains_secure_if_SECURE_COOKIES_is_false(self):
     def test_ignores_if_no_cookies_set(self):
         response = self.client.get('/no_cookie')
         self.assertFalse(response.cookies)
+
+
+class TestSyncUserLanguageMiddleware(SimpleTestCase):
+
+    def setUp(self):
+        self.request = RequestFactory().get('/')
+        self.request.LANGUAGE_CODE = 'en'
+
+    def run_middleware(self):
+        def get_response(request):
+            return HttpResponse('OK')
+
+        middleware = SyncUserLanguageMiddleware(get_response)
+        return middleware(self.request)
+
+    def test_user_language_sets_language_cookie(self):
+        couch_user = FakeUser(language='fra')
+        self.request.couch_user = couch_user
+        response = self.run_middleware()
+
+        language_cookie = response.cookies.get(settings.LANGUAGE_COOKIE_NAME)
+        assert language_cookie.value == couch_user.language
+
+    def test_same_language_does_not_set_cookie(self):
+        couch_user = FakeUser(language=self.request.LANGUAGE_CODE)
+        self.request.couch_user = couch_user
+        response = self.run_middleware()
+
+        language_cookie = response.cookies.get(settings.LANGUAGE_COOKIE_NAME)
+        assert language_cookie is None
```

Comment:
Nit: I feel like this test could be made cleaner by explicitly specifying the request's language code in the test body (i.e. "construct an english request", "create an english user", "verify that no language cookie was set" is easy to follow)